### ISSUE TITLE
gofmt detects error only when ran on explecit file

### ISSUE THREAD

**Issue by xsteadfastx** [golangci/golangci-lint#2228]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

i discovered that vim with ale complained about about file File is not 
```
`gofmt`-ed with `-s` (gofmt)
```
my normal run of golangci-lint doesnt detect that.

```console
$ golangci-lint run --no-config --enable-all --disable=exhaustivestruct,godox
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'.
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.
```

if i run this on the file itself i get the error:

```console
$ golangci-lint run --no-config --enable-all --disable=exhaustivestruct,godox pkg/ftp/ftp_test.go
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'.
pkg/ftp/ftp_test.go:1: File is not `gofmt`-ed with `-s` (gofmt)
// +build integration
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.42.1 built from 54f4301 on 2021-09-04T14:39:09Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
/bin/cat: .golangci.yml: No such file or directory
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.17 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/marv/.cache/go-build"
GOENV="/home/marv/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/marv/.local/share/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/marv/.local/share/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/linuxbrew/.linuxbrew/Cellar/go/1.17/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/linuxbrew/.linuxbrew/Cellar/go/1.17/libexec/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17"
GCCGO="gccgo"
AR="ar"
CC="gcc-5"
CXX="g++-5"
CGO_ENABLED="1"
GOMOD="/home/marv/wip/omsporter/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build954088042=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/marv/wip/omsporter /home/marv/wip /home/marv /home /]
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck]
INFO [loader] Go packages loading at mode 575 (types_sizes|compiled_files|deps|exports_file|files|imports|name) took 156.076822ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 2.086164ms
INFO [linters context/goanalysis] analyzers took 12.387738309s with top 10 stages: buildir: 9.565534738s, inspect: 934.339198ms, fact_deprecated: 346.963515ms, nilness: 328.374265ms, ctrlflow: 294.8648ms, fact_purity: 260.735102ms, printf: 196.662788ms, SA5012: 158.26032ms, typedness: 140.492335ms, structcheck: 27.10327ms
INFO [runner] Issues before processing: 3, after processing: 0
INFO [runner] Processors filtering stat (out/in): skip_files: 3/3, skip_dirs: 3/3, exclude-rules: 0/3, path_prettifier: 3/3, cgo: 3/3, identifier_marker: 3/3, filename_unadjuster: 3/3, autogenerated_exclude: 3/3, exclude: 3/3
INFO [runner] processing took 147.111µs with stages: identifier_marker: 50.17µs, autogenerated_exclude: 31.868µs, path_prettifier: 26.663µs, skip_dirs: 23.887µs, exclude-rules: 7.477µs, cgo: 1.292µs, uniq_by_line: 968ns, nolint: 913ns, filename_unadjuster: 812ns, max_same_issues: 671ns, max_from_linter: 337ns, exclude: 328ns, skip_files: 321ns, source_code: 314ns, severity-rules: 266ns, max_per_file_from_linter: 259ns, diff: 191ns, path_shortener: 138ns, sort_results: 136ns, path_prefixer: 100ns
INFO [runner] linters took 4.673326179s with stages: goanalysis_metalinter: 4.673116553s
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 50 samples, avg is 444.5MB, max is 732.6MB
INFO Execution took 4.836607437s
```

</details>

### Code example or link to a public repository

<details>

```go
// +build integration

// nolint:paralleltest,funlen,gochecknoglobals
package ftp_test
```

</details>

**Comment by ldez** [comment]
Hello,

the constraints must be provided to golangci-lint.

```
  build-tags:
    - integration
```

without that the files are excluded.

**Comment by xsteadfastx** [comment]
hi,

On Sa, 11. Sep 09:06, Ludovic Fernandez wrote:

> the constraints must be provided to golangci-lint.

is there a way to include all build tags?

**Comment by SVilgelm** [comment]
> is there a way to include all build tags?

no, you need to explicitly mention the tags
